tar/multitape/multitape_write.c

```diff
@@ -528,7 +531,8 @@ writetape_open(uint64_t machinenum, const char * cachedir,
 err5:
 	storage_write_free(d->S);
 err4:
-	close(d->lockfd);
+	if (d->lockfd >= 0)
```

Comment:
`!= -1`.  There's only one value which we're using to indicate "cache directory is not locked".  Other negative values would be invalid but we shouldn't hide bugs like that.